### ISSUE TITLE
com.hazelcast.jet.sql.impl.type.BasicNestedFieldsTest

### ISSUE THREAD

**Issue by frant-hartm** [hazelcast/hazelcast#23137]
master (commit b95dbf37be6351208e4b263b97ef827f31d1a3a2)

Failed on PR builder : http://jenkins.hazelcast.com/job/Hazelcast-pr-builder/14821/testReport/junit/com.hazelcast.jet.sql.impl.type/BasicNestedFieldsTest/test_objectComparison_useClient_false_/

https://github.com/hazelcast/hazelcast/pull/23132#issuecomment-1344247405

<details><summary>Stacktrace:</summary>

```
com.hazelcast.sql.HazelcastSqlException: Type already exists: UserType
	at com.hazelcast.sql.impl.QueryUtils.toPublicException(QueryUtils.java:74)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:224)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:175)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:171)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:167)
	at com.hazelcast.sql.SqlService.execute(SqlService.java:89)
	at com.hazelcast.jet.sql.impl.type.BasicNestedFieldsTest.initDefault(BasicNestedFieldsTest.java:519)
	at com.hazelcast.jet.sql.impl.type.BasicNestedFieldsTest.test_objectComparison(BasicNestedFieldsTest.java:126)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: com.hazelcast.sql.impl.QueryException: Type already exists: UserType
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:69)
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:61)
	at com.hazelcast.sql.impl.QueryException.error(QueryException.java:57)
	at com.hazelcast.jet.sql.impl.schema.TableResolverImpl.createType(TableResolverImpl.java:213)
	at com.hazelcast.jet.sql.impl.PlanExecutor.execute(PlanExecutor.java:553)
	at com.hazelcast.jet.sql.impl.SqlPlanImpl$CreateTypePlan.execute(SqlPlanImpl.java:1407)
	at com.hazelcast.sql.impl.SqlServiceImpl.query0(SqlServiceImpl.java:269)
	at com.hazelcast.sql.impl.SqlServiceImpl.execute(SqlServiceImpl.java:207)
	... 18 more
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: test_missingType[useClient:false] in 0.210 seconds.
Started Running Test: test_objectComparison[useClient:false]
12:23:40,171  INFO |test_objectComparison[useClient:false]| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
12:23:40,171  INFO |test_objectComparison[useClient:false]| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
12:23:40,171  INFO |test_objectComparison[useClient:false]| - [SqlTestSupport] Time-limited test - Removing 0 cached plans in SqlTestSupport.@After
12:23:40,172  INFO |test_objectComparison[useClient:false]| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
12:23:40,186  INFO |test_objectComparison[useClient:false]| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog]
BuildInfo right after test_objectComparison[useClient:false](com.hazelcast.jet.sql.impl.type.BasicNestedFieldsTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20221209', buildNumber=20221209, revision=4d16fe4, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'test_objectComparison[useClient:false](com.hazelcast.jet.sql.impl.type.BasicNestedFieldsTest):'
12:23:35, accumulated pauses: 758 ms, max pause: 645 ms, pauses over 1000 ms: 0
12:23:40, accumulated pauses: 0 ms, max pause: 0 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```

</details>

**Comment by olukas** [comment]
Test `com.hazelcast.jet.sql.impl.type.BasicNestedFieldsTest.test_wholeObjectSelect[useClient:false]` failed with the same exception on _master_ with corretto-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-CorrettoJDK8/215/testReport/com.hazelcast.jet.sql.impl.type/BasicNestedFieldsTest/test_wholeObjectSelect_useClient_false_/

**Comment by sumnerib** [comment]
Failed again on _5.3.z_ with corretto-17: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-CorrettoJDK17/65/testReport/junit/com.hazelcast.jet.sql.impl.type/BasicNestedFieldsTest/test_nullValueInRow_useClient_false_/

**Comment by kwart** [comment]
Another PR builder failure (5.3.3): https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/21135/

**Comment by k-jamroz** [comment]
Probably fixed in 5.4 with other UDT changes, see https://github.com/hazelcast/hazelcast/pull/25269#discussion_r1381705152. Will not be backported to 5.3.
